Task25 betti stats

Sondzus opened the pull request:
bettiStats implemented, documentation added, basic tests added. 

The method, as it stands, returns mean Betti shape and mean Betti table. 
There is an option to exclude zero ideals from the list. 


TODO:

- [x] we probably want to add stdev of Betti table; this requires looking at the code where we turn BettyTally into a matrix and simply computing stdev of each entry.
- [x] we also need to decide whether to totally kill the file-writing option (this stores all betti tables computed for a sample; super-useful if large sample) - but we can only do this if we explain how to store everything using Richard's new Sample type object. 
- [x] once 2 is done we can do things like also return % pure tables  -- the code already computes this but doesn't return it (see "--pure" added to return commands in code).

Sondzus committed:
prepared basic bettiStats

sonjapetrovic committed:
deleted output files

sonjapetrovic committed:
removing some file writing leftovers

sonjapetrovic committed:
outlined documentation for bettiStats

sonjapetrovic committed:
wrote documentation and removed file writing for statistics

but left file writing for list of betti tables as an option!!

sonjapetrovic committed:
doc updates

sonjapetrovic committed:
streamlined exclusion of zero ideals from bettiStats

using optional input

sonjapetrovic committed:
SeeAlso added to IncludeZeroIdeals

sonjapetrovic committed:
added comment to doc.

Also note “isPure” is computed but not returned yet. Can’t decide how
to do it prettily - maybe this should be left for the Sample type??

sonjapetrovic committed:
comments deleted/added in method.

sonjapetrovic committed:
added 2 simple tests + minor doc wording change

sonjapetrovic committed:
fixed stuff about excluding zero ideals. much better now.

sonjapetrovic committed:
same as previous

sonjapetrovic committed:
reordered output: (meanBettiShape,meanBetti)

this is so that it makes more sense to append stdDev(betti) to the
output!

Sondzus committed:
Merge branch 'master' into TASK25-bettiStats

sonjapetrovic committed:
unified way to write monomial ideals in 1 line

sonjapetrovic committed:
Merge branch 'master' into TASK25-bettiStats

# Conflicts:
#	RandomMonomialIdeals.m2

Sondzus committed:
Merge branch 'master' into TASK25-bettiStats

Sondzus commented:
I cleaned up the conflicts with base branch; hopefully everything is still good. GitHub can be quite silly and confused at times...

Sondzus commented:
@parkerjoncus  up for helping me figure out how to quickly compute stdev for each matrix entry in this method?

sonjapetrovic committed:
small reordering.

parkerjoncus commented:
@Sondzus yeah I can help! Do you need it done today or can we work on it tomorrow?

sonjapetrovic committed:
trying to get StdDev, not done

some problems remain. Won’t compile as is, so commented out.
Will come back to this later.

Sondzus commented:
@parkerjoncus  if you're doing something else today, that's fine, we can work on it tomorrow. I'm not able to get it all done right this moment; I pushed a commit but it's not working yet. I'll be offline in the afternoon today anyway.  See you tomorrow and we can figure it out.

sonjapetrovic committed:
fixed typo @ - now it loads.

sonjapetrovic committed:
bettiStats now finally has stdDev as part of output.

TO DO: missing test for bettiStats.

Sondzus commented:
missing: tests update to include stddev. @parkerjoncus  :) 

[to find out how to make a test, check out the documentation node for bettiStats. there is an example with 3 ideals and their betti tables, so we can write the test based on, say, the (2,3)-entry of the table.]

Sondzus committed:
Merge branch 'master' into TASK25-bettiStats

sonjapetrovic committed:
Merge branch 'master' into TASK25-bettiStats

sonjapetrovic committed:
missing 1 test for bettiStats - stdev

sonjapetrovic committed:
finished documentation for bettiStats

sonjapetrovic committed:
added optional inputs and documented them

Sondzus committed:
Merge branch 'master' into TASK25-bettiStats

sonjapetrovic committed:
fixed typos in verbose print stmt

ghummel1 committed:
Std Test - not working

sonjapetrovic committed:
adjusted Verbose print stmt

ghummel1 committed:
Finished Tests

ghummel1 committed:
Merge branch 'master' into TASK25-bettiStats

ghummel1 commented:
load, install, check all worked for me!

sonjapetrovic committed:
SaveBettis takes filename instead of Boolean now
